src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -672,6 +678,52 @@ def ok_to_stop(self,
             return HandleCommandResult(-errno.EBUSY, '', warn_message)
         return HandleCommandResult(0, warn_message, '')
 
+    def find_prometheus_credentials(self, multicluster_config: Dict[str, Any], url: str) -> Optional[Dict[str, Any]]:
+        for _, clusters in multicluster_config['config'].items():
+            for cluster in clusters:
+                prometheus_url = cluster.get('prometheus_url')
+                if prometheus_url:
+                    valid_url = prometheus_url.replace("https://", "").replace("http://", "")  # since target URLs are without scheme
+                    if valid_url == url:  # check if the target URL matches with the prometheus URL (without scheme) in the config
+                        return cluster.get('prometheus_access_info')
+        return None
+
+    def get_cluster_credentials(self, targets: List[str]) -> Dict[str, Any]:
+        clusters_credentials = {}
+        multi_cluster_config_str = str(self.mgr.get_module_option_ex('dashboard', 'MULTICLUSTER_CONFIG'))
```

Comment:
Noted @epuertat , we can open a tracker to modify the way we store the multi_cluster config. For now, since we are nearing blocker's only , it would not be suitable to do this big a change because it would affect many areas of the code.

Reply:
Ok, I suggested 2 things:
* Moving the `MULTICLUSTER_CONFIG` to the mon KV store, instead of Module Options.
* Providing a specification of the `MULTICLUSTER_CONFIG`.

The former has obvious backward-compatibility impact, but the latter does not. Based on this discussion, I added a new entry to the [Dashboard Best Practices](https://github.com/ceph/ceph/wiki/Teams-%E2%80%90-Ceph-Dashboard-%E2%80%90-Best-Practices) (point 3.i: don't use dicts for nested/heterogeneous objects).

I'd expect to see in this PR a namedtuple, class or dataclass defining that config structure.

Reply:
Since, this not related and will be out of scope for this PR, let me do this cleanup in a separate PR.

Reply:
@epuertat , opened this new PR - https://github.com/ceph/ceph/pull/59666 for the cleanup. PTAL